kubernetes/cray-rrs/templates/post-upgrade-hook.yaml

```
@@ -0,0 +1,87 @@
+#
+# MIT License
+#
+#  (C) Copyright 2025 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: update-policy-hook
+  annotations:
+    "helm.sh/hook": post-upgrade
+    "helm.sh/hook-delete-policy": hook-succeeded,hook-failed # Clean up the Job resource after execution
+spec:
+  backoffLimit: 4 # Maximum number of retries before considering the Job failed
+  template:
+    spec:
+      serviceAccountName: cray-rrs-service
+      containers:
+      - name: update-policy
+        image: "{{ .Values.kubectl.image.repository }}:{{ .Values.kubectl.image.tag }}"
+        imagePullPolicy: "{{ .Values.kubectl.image.pullPolicy }}"
+        command: ["/bin/bash", "-c"]
+        args:
+        - |
+          set -e
+
+          echo "Fetching secret and decoding YAML..."
+          SECRET_JSON=$(kubectl -n loftsman get secret site-init -o json)
+          ENCODED_YAML=$(echo "$SECRET_JSON" | jq -r '.data["customizations.yaml"]')
+          DECODED_YAML=$(echo "$ENCODED_YAML" | base64 -d)
+          echo "$DECODED_YAML" > /tmp/customizations.yaml
+
+          # Check if rack_resiliency service is present
+          if ! yq r /tmp/customizations.yaml spec.kubernetes.services | grep -Eq '^rack_resiliency:'; then
+            echo "'rack_resiliency' service not found in customizations.yaml. Exiting."
```

Seems like error messages would be better sent to stderr
```suggestion
            echo "'rack_resiliency' service not found in customizations.yaml. Exiting." 1>&2
```